File: src/tui/theme/builtin.ts
import type { ThemeJson } from "./types";

import auraTheme from "./themes/aura.json";
import ayuTheme from "./themes/ayu.json";
import carbonfoxTheme from "./themes/carbonfox.json";
import catppuccinTheme from "./themes/catppuccin.json";
import catppuccinFrappeTheme from "./themes/catppuccin-frappe.json";
import catppuccinMacchiatoTheme from "./themes/catppuccin-macchiato.json";
import cobalt2Theme from "./themes/cobalt2.json";
import cursorTheme from "./themes/cursor.json";
import draculaTheme from "./themes/dracula.json";
import everforestTheme from "./themes/everforest.json";
import flexokiTheme from "./themes/flexoki.json";
import githubTheme from "./themes/github.json";
import gruvboxTheme from "./themes/gruvbox.json";
import kanagawaTheme from "./themes/kanagawa.json";
import lucentOrngTheme from "./themes/lucent-orng.json";
import materialTheme from "./themes/material.json";
import matrixTheme from "./themes/matrix.json";
import mercuryTheme from "./themes/mercury.json";
import monokaiTheme from "./themes/monokai.json";
import nightowlTheme from "./themes/nightowl.json";
import nordTheme from "./themes/nord.json";
import oneDarkTheme from "./themes/one-dark.json";
import opencodeTheme from "./themes/opencode.json";
import orngTheme from "./themes/orng.json";
import osakaJadeTheme from "./themes/osaka-jade.json";
import palenightTheme from "./themes/palenight.json";
import rosepineTheme from "./themes/rosepine.json";
import solarizedTheme from "./themes/solarized.json";
import synthwave84Theme from "./themes/synthwave84.json";
import tokyonightTheme from "./themes/tokyonight.json";
import vercelTheme from "./themes/vercel.json";
import vesperTheme from "./themes/vesper.json";
import zenburnTheme from "./themes/zenburn.json";

export const DEFAULT_THEME_ID = "opencode";

export const DEFAULT_THEMES: Record<string, ThemeJson> = {
  aura: auraTheme,
  ayu: ayuTheme,
  carbonfox: carbonfoxTheme,
  catppuccin: catppuccinTheme,
  "catppuccin-frappe": catppuccinFrappeTheme,
  "catppuccin-macchiato": catppuccinMacchiatoTheme,
  cobalt2: cobalt2Theme,
  cursor: cursorTheme,
  dracula: draculaTheme,
  everforest: everforestTheme,
  flexoki: flexokiTheme,
  github: githubTheme,
  gruvbox: gruvboxTheme,
  kanagawa: kanagawaTheme,
  "lucent-orng": lucentOrngTheme,
  material: materialTheme,
  matrix: matrixTheme,
  mercury: mercuryTheme,
  monokai: monokaiTheme,
  nightowl: nightowlTheme,
  nord: nordTheme,
  "one-dark": oneDarkTheme,
  opencode: opencodeTheme,
  orng: orngTheme,
  "osaka-jade": osakaJadeTheme,
  palenight: palenightTheme,
  rosepine: rosepineTheme,
  solarized: solarizedTheme,
  synthwave84: synthwave84Theme,
  tokyonight: tokyonightTheme,
  vercel: vercelTheme,
  vesper: vesperTheme,
  zenburn: zenburnTheme,
};

export function getBuiltinThemeIds(): string[] {
  return Object.keys(DEFAULT_THEMES);
}

File: crates/frost-core/src/theme/builtin.rs
use crate::theme::types::ThemeJson;
use std::collections::HashMap;

pub const DEFAULT_THEME_ID: &str = "opencode";

macro_rules! load_theme {
    ($themes:expr, $name:expr, $path:expr) => {
        {
            let json: ThemeJson = serde_json::from_str(include_str!($path))
                .expect(concat!("Failed to parse built-in theme: ", $name));
            $themes.insert($name.to_string(), json);
        }
    };
}

pub fn builtin_themes() -> HashMap<String, ThemeJson> {
    let mut themes = HashMap::new();

    load_theme!(themes, "opencode", "../../../../themes/opencode.json");
    load_theme!(themes, "dracula", "../../../../themes/dracula.json");
    load_theme!(themes, "aura", "../../../../themes/aura.json");
    load_theme!(themes, "ayu", "../../../../themes/ayu.json");
    load_theme!(themes, "carbonfox", "../../../../themes/carbonfox.json");
    load_theme!(themes, "catppuccin", "../../../../themes/catppuccin.json");
    load_theme!(themes, "catppuccin-frappe", "../../../../themes/catppuccin-frappe.json");
    load_theme!(themes, "catppuccin-macchiato", "../../../../themes/catppuccin-macchiato.json");
    load_theme!(themes, "cobalt2", "../../../../themes/cobalt2.json");
    load_theme!(themes, "cursor", "../../../../themes/cursor.json");
    load_theme!(themes, "everforest", "../../../../themes/everforest.json");
    load_theme!(themes, "flexoki", "../../../../themes/flexoki.json");
    load_theme!(themes, "github", "../../../../themes/github.json");
    load_theme!(themes, "gruvbox", "../../../../themes/gruvbox.json");
    load_theme!(themes, "kanagawa", "../../../../themes/kanagawa.json");
    load_theme!(themes, "lucent-orng", "../../../../themes/lucent-orng.json");
    load_theme!(themes, "material", "../../../../themes/material.json");
    load_theme!(themes, "matrix", "../../../../themes/matrix.json");
    load_theme!(themes, "mercury", "../../../../themes/mercury.json");
    load_theme!(themes, "monokai", "../../../../themes/monokai.json");
    load_theme!(themes, "nightowl", "../../../../themes/nightowl.json");
    load_theme!(themes, "nord", "../../../../themes/nord.json");
    load_theme!(themes, "one-dark", "../../../../themes/one-dark.json");
    load_theme!(themes, "orng", "../../../../themes/orng.json");
    load_theme!(themes, "osaka-jade", "../../../../themes/osaka-jade.json");
    load_theme!(themes, "palenight", "../../../../themes/palenight.json");
    load_theme!(themes, "rosepine", "../../../../themes/rosepine.json");
    load_theme!(themes, "solarized", "../../../../themes/solarized.json");
    load_theme!(themes, "synthwave84", "../../../../themes/synthwave84.json");
    load_theme!(themes, "tokyonight", "../../../../themes/tokyonight.json");
    load_theme!(themes, "vercel", "../../../../themes/vercel.json");
    load_theme!(themes, "vesper", "../../../../themes/vesper.json");
    load_theme!(themes, "zenburn", "../../../../themes/zenburn.json");

    themes
}
